bayes/naive_bayes.js

```diff
@@ -0,0 +1,324 @@
+StreamModel = require('../core/stream_model.js')
+utils = require('../utils/utils.js')
+
+class BaseNB {
+    /**
+     * Abstract base class for naive Bayes estimators.
+     */
+    constructor() {
+    }
+    predict(X) {
+        /**
+         * Perform classification on an array of test vectors X.
+         * 
+         * @param {array} X
+         * @returns Predicted target values for X.
+         */
+        let jll = this._joint_log_likelihood(X);
+        let cla = [];
+        for(let i = 0; i < jll.length; i++) {
+            for(let j = 0; j < jll[0].length; j++) {
+                if(jll[i][j] == Math.max(...jll[i])) {
+                    cla.push(this.classes_[j]);
+                    break;
+                }
+            }
+        }
+        return (cla);
+    }
+
+    predict_log_proba(X) {
+        /**
```

Comment:
Move description in front of the function. Describe X. Define return of the function in term of type {array} or sonething similar.